src/libraries/System.Private.CoreLib/src/System/Number.Formatting.cs

```diff
@@ -1186,18 +1179,18 @@ private static unsafe void Int32ToNumber(int value, ref NumberBuffer number)
                 value = -value;
             }
 
-            byte* buffer = number.GetDigitsPointer();
-            byte* p = UInt32ToDecChars(buffer + Int32Precision, (uint)value, 0);
+            Span<byte> buffer = number.Digits;
+            int p = UInt32ToDecChars(buffer.Slice(0, Int32Precision), (uint)value, 0);
 
-            int i = (int)(buffer + Int32Precision - p);
+            int i = Int32Precision - p;
 
             number.DigitsCount = i;
             number.Scale = i;
 
-            byte* dst = number.GetDigitsPointer();
+            int dst = 0;
             while (--i >= 0)
-                *dst++ = *p++;
-            *dst = (byte)('\0');
+                buffer[dst++] = buffer[p++];
+            buffer[dst] = (byte)('\0');
```

Comment:
Same comment about `Span.CopyTo`, particularly with this being ~10 for 32-bit integers and ~20 for 64-bit integers.

It may be more directly measurable for `float`/`double` which can be ~113 and ~768 digits in length and where the copy may be much larger.